asserts/snapasserts/validation_sets.go

```diff
@@ -189,6 +194,37 @@ type ValidationSets struct {
 	snaps map[string]*snapContraints
 }
 
+// Revisions returns the set of snap revisions that is enforced by the
+// validation sets that ValidationSets manages.
+func (v *ValidationSets) Revisions() (map[string]snap.Revision, error) {
```

Comment:
nitpick: methods on a struct are usually defined after the any New* function, this and the next method should be defined further down I think